merlin_models/tf/utils/tf_utils.py

```diff
@@ -82,8 +82,7 @@ def maybe_deserialize_keras_objects(
     return config
 
 
-def extract_topk(ks, scores, labels):
-    max_k = tf.reduce_max(ks)
+def extract_topk(max_k, scores, labels):
     topk_scores, topk_indices = tf.math.top_k(scores, max_k)
     topk_labels = gather_torch_like(labels, topk_indices, max_k)
```

Comment:
Maybe later check if we could come up with a more efficient implementation of `gather_torch_like` which does not using `TensorArray`, but rather uses only `tf.gather_nd()` with a 2D index (rows, cols), like the one I implemented here [here](https://github.com/NVIDIA-Merlin/NVTabular/blob/dc9c04d126b8ad842a32a3618e8f4495afbe0678/nvtabular/loader/tensorflow.py#L404). 
Not a priority for now, but maybe it might improve the metrics performance